src/streaming/net/HTTPLoader.js

```diff
@@ -566,9 +569,13 @@ function HTTPLoader(cfg) {
      * @private
      */
     function _updateRequestUrlAndHeaders(request) {
-
-        if (settings.get().streaming.cmcd && settings.get().streaming.cmcd.enabled) {
-            const cmcdMode = settings.get().streaming.cmcd.mode;
+        const currentServiceLocation = request?.serviceLocation;
```

Comment:
Please move the whole CMCD logic to a dedicated function that is called within `_updateRequestUrlAndHeaders`

Reply:
If this is too much work let me know then I do this in a dedicated PR. Thanks